portal/static/js/main.js

```diff
@@ -2112,66 +2112,103 @@ var assembleContent = {
-                                identifier.system != SYSTEM_IDENTIFIER_ENUM["external_site_id"] &&
-                                identifier.system != SYSTEM_IDENTIFIER_ENUM["practice_region"]) {
-                                identifiers.push(identifier);
-                            }
-                        });
-                    }
-                }).fail(function(xhr) {
-                   tnthAjax.reportError(userId, "api/demographics"+userId, xhr.responseText);
-                });
+            var identifiers = null;
+            /*
+             * get current identifier(s)
+             */
+            $.ajax ({
+                type: "GET",
+                url: "/api/demographics/"+userId,
+                async: false
+            }).done(function(data) {
+                if (data && data.identifier) {
+                    identifiers = [];
+                    (data.identifier).forEach(function(identifier) {
+                        identifiers.push(identifier);
+                    });
+                }
+            }).fail(function(xhr) {
+               tnthAjax.reportError(userId, "api/demographics"+userId, xhr.responseText);
+            });
 
-                if (hasStudyId) {
+            /*
+             * NOTE: this will save study Id or site Id only if each has a value
+             * otherwise if each is empty, it will be purged from the identifiers
+             */
+
+            if (hasStudyIdField || hasSiteIdField) {
+                if (hasStudyIdField) {
                     studyId = $.trim(studyId);
-                    var studyIdObj = {
-                        system: SYSTEM_IDENTIFIER_ENUM["external_study_id"],
-                        use: "secondary",
-                        value: studyId
-                    };
+                    if (studyId) {
+                        var studyIdObj = {
+                            system: SYSTEM_IDENTIFIER_ENUM["external_study_id"],
+                            use: "secondary",
+                            value: studyId
+                        };
 
-                    if (identifiers) {
-                        identifiers.push(studyIdObj);
+                        if (identifiers) {
+                            /*
+                             * filter out existing study id as we do not want to save duplicates
+                             */
+                            identifiers = $.grep(identifiers, function(identifier) {
+                                return identifier.system !== SYSTEM_IDENTIFIER_ENUM["external_study_id"];
+                            });
+                            identifiers.push(studyIdObj);
+                        } else {
+                            identifiers = [studyIdObj];
+                        }
                     } else {
-                        identifiers = [studyIdObj];
-                    };
+                        //filter out study ID since it is now empty
```

Comment:
Do you still need to filter these out after Paul's backend changes to remove them?

Reply:
@ivan-c ivan, I think the issue was if the user had a studyid/siteid and later emptied it, we need to purge it so the older studyid/siteid won't be retained - current identifiers can hold the older value of each.

Reply:
Ah, I was wondering if that was the case. I think it might be more appropriate to fix up the bad data with a DB migration than having your code handle every case.
@pbugni 
Do we need a migration for this?